Balanced Distribution, Shared Agents support (#435)
hellohaptik/docs_haptik: docs/agent-chat/chat-assignment.md

3. Chats that are in pending state gets completed when last agent of a team logouts.

4. If an agent is part of two teams, both following Balanced Distribution, the concurrency of that agent will be doubled.

> i.e, If agent concurrency is C, and is a part of N teams that are following Balanced distribution algorithm, his/her concurrency will be C times N.

Reviewer: ```suggestion
i.e, If agent's concurrency is C, and they are a part of N teams that are following Balanced distribution algorithm, their effective concurrency will be C times N.
```

> We recommend keeping a lower agent concurrency for shared agents.

Author: Note

Author: My Chats should show the real correct values for number of chats assigned/pinned as a test case

Author: he = his conc..

Author: If agent concurrency is C, and is a part of N teams following Balanced distribution, he concurrency will be C times N.    ---->>> he can get these many chats actually assigned.

> > Note: We recommend keeping a lower agent concurrency for shared agents.

Reviewer: Should probably add a note stating you can't have agents shared across distribution mode types.

Author: i'll put this point somewhere higher

Author: mentioned at the bottom, under Notes

5. Optimized for: Chat load as it’s almost evenly distributed across online agents at the time.

> 6. Con(s): Unlike Speed Resolution, Quick high performing agents may not be best utilized. 

Reviewer 2: ```suggestion
6. Con(s): Unlike Speed Resolution, high performing agents may not be best utilised. 
```

> 6. Con(s): Does not account for faster agent response times but only the first response times. Faster high performing agents may not be best utilized. 

Reviewer: - Shouldn't this be a continuation of point 5?
 - `Faster high performing agents may not be best utilized.`: seems a bit confusing, not sure what 'faster', 'high performing' mean in this context.

Author: If you have a single client and multiple businesses set up, all businesses must be the same algorithm

--- need to update this for clients with multiple teams and diff algos

Author: changed the wordings by a bit

### Note

> 1. If you have a single client and multiple businesses set up, all businesses must use the same algorithm.

Reviewer 2: ```suggestion
1. If there is a single client and multiple businesses set up, all businesses must use the same algorithm.
```

> 2. If there are multiple clients, one can select different algorithms for each.

Reviewer 2: This sentence does not need to be present as a separate point here, I think @deepwired

Author: 1 - > single client multi business scenarios
2 - > multi client scenario
3 - > agent level stuff

we should keep 3 acc to me, open to re-wording.

Reviewer 2: Alright, rewording it slightly then.

> 2. If there are multiple clients, one can select different algorithms for each.

Reviewer 2: ```suggestion
2. If there are multiple clients, no compulsion of using same algorithm. One can select different algorithms for each.
```
